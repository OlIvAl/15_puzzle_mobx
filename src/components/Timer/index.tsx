import React from 'react';

interface IProps {
  time: string;
}

const Timer: React.FC<IProps> = ({time}) => (
  <div><strong>Time:</strong> {time}</div>
);

export default Timer;