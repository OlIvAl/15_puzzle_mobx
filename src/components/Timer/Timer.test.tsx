import * as React from 'react';
import Timer from '.';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Timer', () => {
  const fakeTime: string = '11:11:11';

  it('он отображается', () => {
    const wrapper = shallow(<Timer time={fakeTime} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('отображает время', () => {
    const wrapper = shallow(<Timer time={fakeTime} />);

    expect(wrapper.contains(fakeTime)).toEqual(true);
  });
});