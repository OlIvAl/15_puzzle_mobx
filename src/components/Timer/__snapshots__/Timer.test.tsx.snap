// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer он отображается 1`] = `
<div>
  <strong>
    Time:
  </strong>
   
  11:11:11
</div>
`;
